src/windows/wslaservice/exe/WSLAVirtualMachine.cpp

```diff
@@ -254,12 +272,20 @@ void WSLAVirtualMachine::Start()
     hvSocketConfig.HvSocketConfig.DefaultConnectSecurityDescriptor = securityDescriptor;
     vmSettings.Devices.HvSocket = std::move(hvSocketConfig);
 
+    CreateVmSavedStateFile();
+    WI_ASSERT(!m_vmSavedStateFile.empty());
```

Comment:
Why do we need to create the file before there is a crash?

Reply:
We have a TempFile helper you'll probably want to use instead to avoid having to manually clean this up.